src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -23,7 +42,7 @@ public static void GetByteCount(Encoding encoding, string chars, int index, int
             Assert.Equal(expected, encoding.GetByteCount(charArray, index, count));
         }
         
-        public static void GetBytes(Encoding encoding, string source, int index, int count, byte[] bytes, int byteIndex, byte[] expectedBytes)
+        private static void GetBytes(Encoding encoding, string source, int index, int count, byte[] bytes, int byteIndex, byte[] expectedBytes)
```

Comment:
what we need is 
GetBytes(Encoding encoding, string source, int index, int count, byte[] expectedBytes)

and just let this method take care of the allocation and testing the results. this will simplify the calling code and in same time we'll have only one place take care of any test of the output buffers